src/replication.c

```diff
@@ -949,11 +949,14 @@ void syncCommand(client *c) {
 
     /* Try a partial resynchronization if this is a PSYNC command.
      * If it fails, we continue with usual full resynchronization, however
-     * when this happens replicationSetupSlaveForFullResync will replied
-     * with:
+     * when this happens masterTryPartialResynchronization() will not
+     * reply with:
      *
      * +FULLRESYNC <replid> <offset>
      *
+     * The reply is actually delayed to replicationSetupSlaveForFullResync(),
+     * which is called immediately after BGSAVE child process forked or reused.
```

Comment:
that's not always correct.
in some cases syncCommand immediately replies with +FULLSYNC